harness/determined/experimental/inference/__init__.py

```diff
@@ -0,0 +1,6 @@
+from determined.experimental.inference._torch_batch_process import (
```

Comment:
I'm not quite sure what's the best solution, but there is a problem with having a `det.experimental.inference` namespace that imports `Torch`.  Having an object which relies on torch like `TorchBatchProcessor` in the `inference` means that nothing in `inference` could ever be imported if torch wasn't available, even if you wanted to add something like a `det.exeprimental.inference.KerasBatchProcessor`.

Where would the ultimate resting place of `TorchBatchProcessor` be?  `det.pytorch.TorchBatchProcessor` perhaps?

So perhaps it should live at `det.pytorch.experimental.TorchBatchProcessor` for now?